### PR TITLE
Server options as functions for passing custom URIs and AllowOrigins easily

### PR DESCRIPTION
**Pull request by yasarix**
I'd like to thank you for this package. It is really hard to find an existing implementation that works with App Engine,

I saw that you are using hardcoded `/verify` and `/token` URIs for handlers, and you had a TODO tag for passing custom `AllowOrigins`. Since I needed custom URIs for handlers and also custom AllowOrigins, I made a small modification in server.go to add these.

I added an optional `options` parameter to `auth.Server()` method. `options` are list of functions that you can pass to set options:
- `ServerGenerateURI` to pass a custom URI for s.generateHandler
- `ServerVerifyURI` to pass a custom URI for s.verifyHandler
- `ServerAllowedOrigins` to pass a custom list of origins into AllowedOrigins

In case any these are not passed, defaults are set as they were in original implementation.

If no options passed, `auth.Server()` will run as it was, so any existing implementation won't be broken.

I also modified sample app.go to show these options used:
`
http.Handle("/", auth.Server(customClaims,
	// Custom AllowedOrigins
	firebase.ServerAllowedOrigins([]string{"*"}),
	// Custom URI for verifyHandler
	firebase.ServerVerifyURI("/customVerify"),
	// Custom URI for generateHandler
	firebase.ServerGenerateURI("/customGenerate")))
`

I hope you find this merge request useful and accept it.

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -17,7 +17,13 @@ func init() {
 	auth = fb.Auth()
 
 	// auth server comes with CORS included
-	http.Handle("/", auth.Server(customClaims))
+	http.Handle("/", auth.Server(customClaims,
+		// Custom AllowedOrigins
+		firebase.ServerAllowedOrigins([]string{"*"}),
+		// Custom URI for verifyHandler
+		firebase.ServerVerifyURI("/customVerify"),
```
Comment (CaptainCodeman) [inline app/app.go]: I like showing how they are set, but could they be set back to the default string even though it seems redundant - the Polymer auth client lib uses this deployed example for it's demo.

**Comment by CaptainCodeman** [comment]
Thanks for the changes, I'm a fan of the function config approach in Go :thumbsup:

Just that one minor change to align with the separate [Polymer lib demo](https://github.com/CaptainCodeman/auth-ajax/blob/master/demo/firebase-demo.html) and I'll merge this in.

File: app/app.go (hunk 1 above)
Comment (yasarix) [reply app/app.go]: OK, I will make an update to keep the default URIs here.

**Comment by yasarix** [comment]
Thanks. I recently started using it, and it is an elegant approach.

I made the changes that you have requested.

**Comment by CaptainCodeman** [comment]
Thanks for the PR @yasarix !

**Pull request merged**